tests/test_minimal/test_utils/test_dict.py

```diff
@@ -0,0 +1,36 @@
+import unittest
+
+from neuroconv.utils.dict import DeepDict
+
+
+class TestDeepDict(unittest.TestCase):
+    def setUp(self):
+        self.dd = DeepDict()
+        self.dd["a"]["b"]["c"] = 42
+        self.data = {"a": {"b": {"c": 42}}}
+
+    def test_getitem(self):
+        self.assertEqual(self.dd["a"]["b"]["c"], 42)
+
+    def test_missing_key(self):
+        dd = DeepDict()
+        self.assertIsInstance(dd["non_existent"], DeepDict)
+
+    def test_to_dict(self):
+        expected = self.data
+        self.assertEqual(self.dd.to_dict(), expected)
```

Comment:
Is this equivalent to `dict(my_deep_dict)`? I assume that would be the more common calling case - I also assumed it would need to be called `__to_dict__` or something for that to work

Reply:
Upon inspecting the `OrderedDict` class, it's just called `__dict__`. Not a bad thing to keep `.to_dict()` as is, but I'd recommend adding `__dict__` and a test that `dict(my_deep_dict)` works for convenience

Reply:
ok, done

Reply:
> ok, done

Not seeing this in the code or tests

Reply:
Oh, nevermind, I see the test now

Reply:
The `__dict__` magic method is removed. `DeepDict.to_dict()`was working properly, but `dict(DeepDict)` was not being applied recursively. Apparently `dict()` does not call `__dict__` in this case? I couldn't quite figure it out.